utils/python/CIME/case.py

```diff
@@ -491,9 +491,10 @@ def configure(self, compset_name, grid_name, machine_name=None,
                       'COMPILER' not in x and 'MPILIB' not in x]
 
         for nodename in nodenames:
-            value = machobj.get_value(nodename)
+            value = machobj.get_value(nodename, resolved=False)
```

Comment:
This resolved=False causes the EXEROOT and other variables in the file to be $CESMSCRATHROOT/$CASE/bld
instead of /glade/scratch/jedwards/whatever/bld  without the resolved=False this was trying to resolve case before the case variable is defined in xml and so was getting it from the environment.